### PR TITLE
XUnit conversion

### PR DESCRIPTION
**Pull request by JakeGinnivan**
And switched all assertions over to shouldly.

NUnit is overcomplicated and promotes bad practices (which we are trying to avoid at teststack). I think we should be using a better unit test framework.

### CODE REVIEW

**Comment by mwhelan** [comment]
Why the downgrade to xunit 1.9.2?

**Comment by JakeGinnivan** [comment]
https://github.com/approvals/ApprovalTests.Net/issues/107

**Comment by JakeGinnivan** [comment]
Getting failing tests..

```
System.Configuration.ConfigurationErrorsException : Configuration system failed to initialize
---- System.Configuration.ConfigurationErrorsException : Data at the root level is invalid. Line 1, position 1. (C:\TeamCity\buildAgent\work\94ce9f3686219309\Build\Test\TestStack.BDDfy.Samples.dll line 1)
-------- System.Xml.XmlException : Data at the root level is invalid. Line 1, position 1.
   at System.Configuration.ConfigurationManager.PrepareConfigSystem()
   at System.Configuration.ConfigurationManager.GetSection(String sectionName)
   at System.Web.Configuration.ClientRuntimeConfig.GetSectionObject(String sectionName)
   at System.Web.Configuration.RuntimeConfig.GetSection(String sectionName, Type type, ResultsIndex index)
   at System.Web.Configuration.RuntimeConfig.get_HttpRuntime()
   at System.Web.Util.HttpEncoder.GetCustomEncoderFromConfig()
   at System.Lazy`1.CreateValue()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Lazy`1.get_Value()
   at System.Web.Util.HttpEncoder.get_Current()
   at TestStack.BDDfy.Reporters.Html.MetroReportBuilder.AddScenario(Scenario scenario) in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy\Reporters\Html\MetroReportBuilder.cs:line 290
   at TestStack.BDDfy.Reporters.Html.MetroReportBuilder.AddScenario(Scenario[] scenarioGroup) in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy\Reporters\Html\MetroReportBuilder.cs:line 194
   at TestStack.BDDfy.Reporters.Html.MetroReportBuilder.AddStory(Story story) in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy\Reporters\Html\MetroReportBuilder.cs:line 182
   at TestStack.BDDfy.Reporters.Html.MetroReportBuilder.ResultDetails() in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy\Reporters\Html\MetroReportBuilder.cs:line 146
   at TestStack.BDDfy.Reporters.Html.MetroReportBuilder.HtmlBody() in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy\Reporters\Html\MetroReportBuilder.cs:line 60
   at TestStack.BDDfy.Reporters.Html.MetroReportBuilder.CreateReport(HtmlReportModel model) in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy\Reporters\Html\MetroReportBuilder.cs:line 34
   at TestStack.BDDfy.Reporters.Html.MetroReportBuilder.TestStack.BDDfy.Reporters.IReportBuilder.CreateReport(FileReportModel model) in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy\Reporters\Html\MetroReportBuilder.cs:line 24
   at TestStack.BDDfy.Tests.Reporters.ReportApprover.Approve(Func`1 model, IReportBuilder reportBuilder) in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy.Tests\Reporters\ReportApprover.cs:line 17
   at TestStack.BDDfy.Tests.TagsTests.TagsAreReportedInMetroHtmlReport() in c:\TeamCity\buildAgent\work\94ce9f3686219309\TestStack.BDDfy.Tests\TagsTests.cs:line 57
----- Inner Stack Trace -----
   at System.Configuration.ConfigurationSchemaErrors.ThrowIfErrors(Boolean ignoreLocal)
   at System.Configuration.BaseConfigurationRecord.ThrowIfParseErrors(ConfigurationSchemaErrors schemaErrors)
   at System.Configuration.BaseConfigurationRecord.ThrowIfInitErrors()
   at System.Configuration.ClientConfigurationSystem.EnsureInit(String configKey)
----- Inner Stack Trace -----
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.Throw(String res, String arg)
   at System.Xml.XmlTextReaderImpl.ParseRootLevelWhitespace()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlTextReaderImpl.Read()
   at System.Xml.XmlTextReader.Read()
   at System.Configuration.XmlUtil..ctor(Stream stream, String name, Boolean readToFirstElement, ConfigurationSchemaErrors schemaErrors)
   at System.Configuration.BaseConfigurationRecord.InitConfigFromFile()
------- Stdout: -------

Scenario: Tags are reported in metro html report
    Given a step

Tags: Tag1, Tag 2
```

Anyone got any ideas?

File: Samples/TestStack.BDDfy.Samples/ModuleInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using TestStack.BDDfy.Configuration;
+using TestStack.BDDfy.Reporters.Html;
+using TestStack.BDDfy.Samples;
+using TestStack.BDDfy.Samples.Atm;
+
+/// <summary>
+/// Used by the ModuleInit. All code inside the Initialize method is ran as soon as the assembly is loaded.
+/// </summary>
+public static class ModuleInitializer
```
Comment (MehdiK) [inline Samples/TestStack.BDDfy.Samples/ModuleInitializer.cs]: I really wish xUnit would add this really critical missing bit. It is VERY annoying.

File: Samples/TestStack.BDDfy.Samples/TestStack.BDDfy.Samples.csproj [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\..\packages\xunit.runner.visualstudio.2.0.0-rc1-build1030\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\..\packages\xunit.runner.visualstudio.2.0.0-rc1-build1030\build\net20\xunit.runner.visualstudio.props')" />
```
Comment (MehdiK) [inline Samples/TestStack.BDDfy.Samples/TestStack.BDDfy.Samples.csproj]: Hmmm, I've been unsure about adding the VS runner; but I think it won't hurt.

File: Samples/TestStack.BDDfy.Samples/TicTacToe/WinnerGame.cs [hunk 3]
```diff
@@ -38,7 +39,7 @@ IEnumerable<string> GivenTheFollowingBoard()
         IEnumerable<string> ThenTheWinnerShouldBe()
         {
             yield return "Then the winner is " + _expectedWinner;
-            Assert.AreEqual(Game.Winner, _expectedWinner);
+            _expectedWinner.ShouldBe(Game.Winner);
```
Comment (powerdude) [inline Samples/TestStack.BDDfy.Samples/TicTacToe/WinnerGame.cs]: this, and others, should be in the form actual.ShouldBe(expected)

**Comment by MehdiK** [comment]
This is great @JakeGinnivan. Thanks.

File: Samples/TestStack.BDDfy.Samples/TicTacToe/TicTacToe.cs [hunk 4]
```diff
@@ -38,12 +39,12 @@ void ThenTheBoardStateShouldBe(string[] firstRow, string[] secondRow, string[] t
 
         void ThenTheWinnerShouldBe(string expectedWinner)
         {
-            Assert.AreEqual(Game.Winner, expectedWinner);
+            expectedWinner.ShouldBe(Game.Winner);
```
Comment (powerdude) [inline Samples/TestStack.BDDfy.Samples/TicTacToe/TicTacToe.cs]: this, and others, should be in the form actual.ShouldBe(expected)

File: TestStack.BDDfy.Tests/Configuration/StepExecutorTests.cs [hunk 5]
```diff
@@ -56,7 +55,7 @@ public void CustomizingStepExecutionByOverridingStepExecutor()
 ".Replace("\r", string.Empty).Trim();
 
                 string actual = testStepExecutor.Results.Replace("\r", string.Empty).Trim();
-                Assert.AreEqual(expected, actual);
+                expected.ShouldBe(actual);
```
Comment (powerdude) [inline TestStack.BDDfy.Tests/Configuration/StepExecutorTests.cs]: this, and others, should be in the form actual.ShouldBe(expected)

File: TestStack.BDDfy.Tests/Processors/TestRunnerTests.cs [hunk 6]
```diff
@@ -31,7 +31,7 @@ public void InitializesScenarioWithExampleBeforeRunning()
 
             sut.Process(story);
 
-            Assert.AreEqual(expectedValue, actualValue);
+            expectedValue.ShouldBe(actualValue);
```
Comment (powerdude) [inline TestStack.BDDfy.Tests/Processors/TestRunnerTests.cs]: this, and others, should be in the form actual.ShouldBe(expected)

File: Samples/TestStack.BDDfy.Samples/TicTacToe/TicTacToe.cs (hunk 4 above)
Comment (MehdiK) [inline Samples/TestStack.BDDfy.Samples/TicTacToe/TicTacToe.cs]: Good catch

File: Samples/TestStack.BDDfy.Samples/TestStack.BDDfy.Samples.csproj (hunk 2 above)
Comment (JakeGinnivan) [inline Samples/TestStack.BDDfy.Samples/TestStack.BDDfy.Samples.csproj]: There is no longer a xUnit VS Runner extension. This is the recommended way to add support

Comment (JakeGinnivan) [inline Samples/TestStack.BDDfy.Samples/TestStack.BDDfy.Samples.csproj]: As far as I know anyways, things might have changed since I read that

**Comment by JakeGinnivan** [comment]
@MehdiK @powerdude fixed the assertions that were the wrong way around. thanks for reviewing.

**Pull request merged**